### ISSUE TITLE
Mount task skipping

### ISSUE THREAD

**Issue by fdanielson** [ansible/ansible-modules-core#1577]
##### ISSUE TYPE

Bug Report
##### COMPONENT NAME

mount
##### ANSIBLE VERSION

1.9.1
##### SUMMARY

I'm having an issue with a mount task to add an NFS target. I wasn't able to find any real documentation on doing this, so I'm not sure if it's supported at all. The mount command doesn't fail, or provide an error even with -vvvv so I'm not sure what it's doing or what's going wrong.

I'm running Ansible 1.9.1

Here is the task:
- mount: "name=/var/shared_files src='<Server IP>:/var/shared_files' fstype=nfs opts='defaults,noatime,_netdev' state=present"

I have also tried calling it as an action instead of mount, but the results were the same.

Here is the output:
TASK: [webservers | mount name=/var/shared_files src='<Server IP>:/var/shared_files' fstype=nfs opts='defaults,noatime,_netdev' state=present] ***
<server01> ESTABLISH CONNECTION FOR USER: root
<server01> REMOTE_MODULE mount name=/var/shared_files src='<Server IP>:/var/shared_files' fstype=nfs opts='defaults,noatime,_netdev' state=present CHECKMODE=True
<server01> EXEC ['ssh', '-C', '-tt', '-q', '-o', 'ControlMaster=auto', '-o', 'ControlPersist=60s', '-o', 'ControlPath=/root/.ansible/cp/ansible-ssh-%h-%p-%r', '-o', 'KbdInteractiveAuthentication=no', '-o', 'PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey', '-o', 'PasswordAuthentication=no', '-o', 'ConnectTimeout=10', 'server01', "/bin/sh -c 'mkdir -p $HOME/.ansible/tmp/ansible-tmp-1434558002.33-10305240023730 && echo $HOME/.ansible/tmp/ansible-tmp-1434558002.33-10305240023730'"]
<server02> ESTABLISH CONNECTION FOR USER: root
<server02> REMOTE_MODULE mount name=/var/shared_files src='<Server IP>:/var/shared_files' fstype=nfs opts='defaults,noatime,_netdev' state=present CHECKMODE=True
<server02> EXEC ['ssh', '-C', '-tt', '-q', '-o', 'ControlMaster=auto', '-o', 'ControlPersist=60s', '-o', 'ControlPath=/root/.ansible/cp/ansible-ssh-%h-%p-%r', '-o', 'KbdInteractiveAuthentication=no', '-o', 'PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey', '-o', 'PasswordAuthentication=no', '-o', 'ConnectTimeout=10', 'server02', "/bin/sh -c 'mkdir -p $HOME/.ansible/tmp/ansible-tmp-1434558002.34-226129697900819 && echo $HOME/.ansible/tmp/ansible-tmp-1434558002.34-226129697900819'"]
<server01> PUT /tmp/tmppujJbA TO /root/.ansible/tmp/ansible-tmp-1434558002.33-10305240023730/mount
<server02> PUT /tmp/tmpWlK9Sh TO /root/.ansible/tmp/ansible-tmp-1434558002.34-226129697900819/mount
<server02> EXEC ['ssh', '-C', '-tt', '-q', '-o', 'ControlMaster=auto', '-o', 'ControlPersist=60s', '-o', 'ControlPath=/root/.ansible/cp/ansible-ssh-%h-%p-%r', '-o', 'KbdInteractiveAuthentication=no', '-o', 'PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey', '-o', 'PasswordAuthentication=no', '-o', 'ConnectTimeout=10', 'server02', "/bin/sh -c '/usr/bin/python /root/.ansible/tmp/ansible-tmp-1434558002.34-226129697900819/mount; rm -rf /root/.ansible/tmp/ansible-tmp-1434558002.34-226129697900819/ >/dev/null 2>&1'"]
<server01> EXEC ['ssh', '-C', '-tt', '-q', '-o', 'ControlMaster=auto', '-o', 'ControlPersist=60s', '-o', 'ControlPath=/root/.ansible/cp/ansible-ssh-%h-%p-%r', '-o', 'KbdInteractiveAuthentication=no', '-o', 'PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey', '-o', 'PasswordAuthentication=no', '-o', 'ConnectTimeout=10', 'server01', "/bin/sh -c '/usr/bin/python /root/.ansible/tmp/ansible-tmp-1434558002.33-10305240023730/mount; rm -rf /root/.ansible/tmp/ansible-tmp-1434558002.33-10305240023730/ >/dev/null 2>&1'"]

**Comment by verm666** [comment]
1. Why are you skip nfs server ip address in your playbook?
2. state=present only add the new line to /etc/fstab, for real mount (and add line to /etc/fstab obviously)- you should use state=mounted

**Comment by fdanielson** [comment]
Hi Verm,
I just omitted the IP from the post, no real reason.
I tried with state=mounted, state=unmounted, and state=present and all of them skipped. I ran them it on servers that had the drive mounted already, as well as ones that didn't.

**Comment by verm666** [comment]
Hmm...

Can you repeat next steps:
1. Remove related to nfs lines from /etc/fstab
2. Show me your /etc/fstab
3. Run playbook with mount task (with state=mounted) and provide minimal piece of code, which you are use
4. Show 'mount' and 'which mount.nfs' output and /etc/fstab content again

And provide information about you OS.

**Comment by fdanielson** [comment]
Hey sorry, didn't check my messages for a while there.
The target machine is CentOS 6.5
I actually tried that with no effect, either way going from trying to remove it or add it with it there and without.
I didn't explicitly say before that I was running this in check mode, because I wanted to make sure it was actually going to do what I wanted. At some point that I didn't notice, the output of the Playbook changed to this, and offers an explanation.

> skipping: [web.master]
> ok: [web.master] => {"changed": false, "msg": "remote module does not support check mode", >"skipped": true}

**Comment by verm666** [comment]
Hmm... In check mode Ansible obviously doesn't start modules, which doesn't support check mode. More over, in check mode (if module is correct) Ansible doesn't change something on your server.

So, if you tried start mount nfs task with `--check` - task was just skipped and it's ok.

I rechecked mount task with nfs on CentOS, everything working fine.

So, did I answer to your question, @TurtleTechnologies?

**Comment by ansibot** [comment]
@grossws, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/jctanner/ansibullbot/blob/ISSUE_TRIAGE_CLASS/ISSUE_HELP.md)

**Comment by grossws** [comment]
@ansibot, wontfix: irrelevant for ansible 2.0 and 2.1 branches.

**Comment by grossws** [comment]
@gregdek, @jctanner should bot remove `waiting_on_*` in case issue is closed? Also similar case is `needs_contributor`.
